File: examples/run_snapshot.py
"""Call the paid Actor toninovo/google-shopping-price-intelligence. Docs example only."""

import os

from apify_client import ApifyClient

token = os.environ.get("APIFY_TOKEN")
if not token:
    raise SystemExit("Set APIFY_TOKEN before running.")

client = ApifyClient(token)
run = client.actor("toninovo/google-shopping-price-intelligence").call(
    run_input={
        "queries": ["wireless earbuds"],
        "country": "us",
        "language": "en",
        "limit": 10,
        "maxAttempts": 2,
    },
    max_total_charge_usd=0.10,
)
for item in client.dataset(run["defaultDatasetId"]).iterate_items():
    print(
        item.get("title"),
        item.get("priceNumeric"),
        item.get("currency"),
        item.get("merchant"),
        item.get("link"),
        item.get("queryMinPrice"),
        item.get("queryMedianPrice"),
        item.get("queryMaxPrice"),
    )

File: examples/run_snapshot.mjs
// Call the paid Actor toninovo/google-shopping-price-intelligence.
// npm install apify-client
import { ApifyClient } from "apify-client";

const token = process.env.APIFY_TOKEN;
if (!token) {
  throw new Error("Set APIFY_TOKEN before running.");
}

const client = new ApifyClient({ token });
const run = await client.actor("toninovo/google-shopping-price-intelligence").call(
  {
    queries: ["wireless earbuds"],
    country: "us",
    language: "en",
    limit: 10,
    maxAttempts: 2,
  },
  { maxTotalChargeUsd: 0.1 },
);

const { items } = await client.dataset(run.defaultDatasetId).listItems();
for (const item of items) {
  console.log(
    item.title,
    item.priceNumeric,
    item.currency,
    item.merchant,
    item.link,
    item.queryMinPrice,
    item.queryMedianPrice,
    item.queryMaxPrice,
  );
}
